SparkPost/2web2ui
# SA-1849 - (UI) Compare By - Bounce Report

[SA-1849](https://sparkpost.atlassian.net/browse/SA-1849)

### What Changed

- Adds new logic to handle dynamic tab rendering when there are active comparisons
- Adds new custom `usePrepareQuery` to prepare deliverability requests based on current report builder context - updates `CompareByAggregatedMetrics` to use this hook.
- Adds new `getComparisonArguments` helper
- Renames table components within `src/pages/reportBuilder/components/tabs`

### How To Test

- Go to report builder with the compare by UI flag enabled
- Add several comparisons without changing metrics
- Verify the default "Bounce Reason" tab does not render
- Verify that a tab for each comparison renders and makes the correct requests - _I could use some help with this - I think I did this right?_

### Questions

- **Should the tabs logic be extracted in to a custom hook?** This should be feasible, except for the unusual `setShowTable` side effect baked-in. This could also be baked in to report builder context to centralize things? Or we just don't worry about it until we do our refactoring in Q1.
- **Should `usePrepareQuery` have a different/more-specific name?**

### To Do

- [x] Handle API errors by adding relevant content and retries
- [x] Add supporting tests for API errors
- [x] Rename API helpers in `src/helpers/api` and remove imports in to the root index file
- [x] Rename existing components within `src/pages/reportBuilder/tabs` to be consistent with `BounceReportTab`
- [X] Ensure "Count %" is rendering correctly in the table - doesn't appear to be at the moment.
